lua/tests/infoview/run_spec.lua

```diff
@@ -9,39 +9,94 @@ require('tests.helpers').setup {
+      assert.is.equal(num_wins, get_num_wins())
+    end)
+
+    it('opens',
+    function(_)
+      local num_wins = get_num_wins()
+      infoview_info = infoview.open()
+      assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
+      assert.is.equal(num_wins + 1, get_num_wins())
+    end)
 
-  it('opens',
-  function(_)
-    local num_wins = get_num_wins()
-    infoview_info = infoview.open()
-    assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
-    assert.is.equal(num_wins + 1, get_num_wins())
+    it('remains open on BufWinEnter',
+    function(_)
+      local num_wins = get_num_wins()
+      vim.api.nvim_command("edit lua/tests/fixtures/example-lean3-project/test/test1.lean")
+      assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
+      assert.is.equal(num_wins, get_num_wins())
+    end)
   end)
 
-  it('remains open on BufWinEnter',
-  function(_)
-    infoview_info = infoview.open()
-    local num_wins = get_num_wins()
-    vim.api.nvim_command("edit lua/tests/fixtures/example-lean3-project/test/test1.lean")
-    assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
-    assert.is.equal(num_wins, get_num_wins())
+  infoview_info = infoview.open()
+
+  vim.api.nvim_command("tabedit lua/tests/fixtures/example-lean4-project/Test.lean")
+
+  local new_win = vim.api.nvim_get_current_win()
+  local new_infoview_info = infoview.open()
+
+  describe('new tab', function()
+    it('closes independently',
+    function(_)
+      local num_wins = get_num_wins()
+      infoview.close()
+      assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
+      assert.is_false(vim.api.nvim_win_is_valid(new_infoview_info.window))
+      assert.is.equal(num_wins - 1, get_num_wins())
+    end)
+
+    it('remains closed on BufWinEnter',
+    function(_)
+      local num_wins = get_num_wins()
+      vim.api.nvim_command("edit lua/tests/fixtures/example-lean4-project/Test/Test1.lean")
+      assert.is.equal(num_wins, get_num_wins())
+    end)
+
+    it('opens independently',
+    function(_)
+      vim.api.nvim_set_current_win(win)
+      infoview.close()
+      vim.api.nvim_set_current_win(new_win)
+      local num_wins = get_num_wins()
+      new_infoview_info = infoview.open()
+      assert.is_false(vim.api.nvim_win_is_valid(infoview_info.window))
+      assert.is_true(vim.api.nvim_win_is_valid(new_infoview_info.window))
+      assert.is.equal(num_wins + 1, get_num_wins())
+    end)
+
+    it('remains open on BufWinEnter',
+    function(_)
+      local num_wins = get_num_wins()
+      vim.api.nvim_command("edit lua/tests/fixtures/example-lean4-project/Test/Test1.lean")
+      assert.is_true(vim.api.nvim_win_is_valid(new_infoview_info.window))
+      assert.is.equal(num_wins, get_num_wins())
+    end)
+
+    vim.api.nvim_command("tabnew")
+    pending('opens automatically after having closen previous infoviews',
```

Comment:
Currently fails because the `LeanInfoview` autocmds are cleared after a close, so for now you'll have to manually open on new tab if you want an infoview. Dynamically configurable infoview auto-open coming in the next PR.